packages/attribute-registry-process/test/api/getAttributeById.test.ts

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { describe, it, expect, vi } from "vitest";
+import request from "supertest";
+import jwt from "jsonwebtoken";
+import { Attribute, generateId } from "pagopa-interop-models";
+import {
+  createPayload,
+  getMockAttribute,
+  getMockAuthData,
+  getSystemOrUserAuthData,
+} from "pagopa-interop-commons-test";
+import { AuthData, systemRole, userRole } from "pagopa-interop-commons";
+import { attributeRegistryApi } from "pagopa-interop-api-clients";
+import { api } from "../vitest.api.setup.js";
+import { toApiAttribute } from "../../src/model/domain/apiConverter.js";
+import { attributeRegistryService } from "../../src/routers/AttributeRouter.js";
+import { attributeNotFound } from "../../src/model/domain/errors.js";
+
+describe("API /attributes/{attributeId} authorization test", () => {
+  const attribute: Attribute = getMockAttribute();
+
+  const apiAttribute = attributeRegistryApi.Attribute.parse(
+    toApiAttribute(attribute)
+  );
+
+  vi.spyOn(attributeRegistryService, "getAttributeById").mockResolvedValue({
+    data: attribute,
+    metadata: { version: 1 },
+  });
+  const generateToken = (authData: AuthData) =>
+    jwt.sign(createPayload(authData), "test-secret");
+
+  const makeRequest = async (token: string, attributeId: string) =>
+    request(api)
+      .get(`/attributes/${attributeId}`)
+      .set("Authorization", `Bearer ${token}`)
+      .set("X-Correlation-Id", generateId())
+      .send();
+
+  it.each([
+    userRole.ADMIN_ROLE,
+    userRole.API_ROLE,
+    userRole.SECURITY_ROLE,
+    systemRole.M2M_ROLE,
+    userRole.SUPPORT_ROLE,
+  ])("Should return 200 for user with role %s", async (role) => {
+    const token = generateToken(getSystemOrUserAuthData(role));
+    const res = await makeRequest(token, attribute.id);
+    expect(res.status).toBe(200);
+    expect(res.body).toEqual(apiAttribute);
+  });
+
+  it.each([
+    systemRole.INTERNAL_ROLE,
+    systemRole.MAINTENANCE_ROLE,
+    systemRole.M2M_ADMIN_ROLE,
```

Comment:
Use here as well a filter on Object.values(AuthRole) so that it also consider future roles that may be added